File: gin-backend/pkg/common/models/company.go
package models

import "gorm.io/gorm"

type Company struct {
	gorm.Model
	Comp_name         string
	Comp_address      string
	Comp_phone        string
	Comp_ruc          string
	Comp_mail         string
	CompanyID         *uint
	Children_comp     []*Company `gorm:"foreignKey:CompanyID"`
	Parent_comp       *Company
	Company_details   []Company_detail
	Devices           []Device
	Users             []Users
	Inventory_headers []Inventory_header
}

File: angular-frontend/src/app/demo/api/company.ts
import { Company_detail } from "./company_detail"
import { Device } from "./device"
import { Inventory_header } from "./inventory_header"
import { User } from "./users"

export interface Company {
    ID?: number,
    Comp_name?: string
    Comp_address?: string
    Comp_mail?: string
    Comp_ruc?: string
    Comp_phone?: string
    CompanyID?: number
    Children_comp?: Company[]
    Company_details?: Company_detail[]
    Devices?: Device[]
    User?: User[]
    Inventory_headers?: Inventory_header[]

}